Drop SendGrid email transport support

florian-strasser opened the pull request:
Based on issue [#23217](https://github.com/directus/directus/issues/23217) I removed the nodemailer-sendgrid else if branch

What's changed:

- We are dropping support sendgrid

## Potential Risks / Drawbacks

- It's a breaking change that forces users that rely on this to switch to SMTP

## Review Notes / Questions

- nodemailer/sendgrid received the last update 6 years ago which should make clear that this absolutely outdated: [https://github.com/nodemailer/nodemailer-sendgrid](https://github.com/nodemailer/nodemailer-sendgrid)

---

Closes #23217

florian-strasser committed:
Drop nodemailer-sendgrid support

Based on issue #23217 I removed the nodemailer-sendgrid else if branch

McSundae commented:
I would also remove the package `api/package.json`, but I'm not sure if it is currently used in any other file than `api/src/mailer.ts`

florian-strasser committed:
Create hot-actors-lay.md

florian-strasser committed:
Remove nodemailer-sendgrid

Since the package received the last update 6 years ago we remove nodemailer-sendgrid for security reasons

McSundae commented:
Not sure how to fix the current errors, probably I should not just remove the line from package.json.

 ` WARN  There are cyclic workspace dependencies: /home/runner/work/directus/directus/api, /home/runner/work/directus/directus/directus` 

`ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with api/package.json`

paescuj commented:
> Not sure how to fix the current errors, probably I should not just remove the line from package.json.
> 
>  ` WARN  There are cyclic workspace dependencies: /home/runner/work/directus/directus/api, /home/runner/work/directus/directus/directus` 
> 
> `ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with api/package.json`

Heya @McSundae 👋 You need to run `pnpm install` to update the lockfile after the change to the `package.json` file.

paescuj commented on .changeset/hot-actors-lay.md:
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/api": patch
+---
+
+Drop nodemailer-sendgrid support since the package is no longer maintained
```
This is a major change to the `@directus/api` package

```suggestion
---
"@directus/api": major
---

Dropped support for the SendGrid email transport option
```

McSundae replied:
As suggested I made the changes

paescuj reviewed (changes requested):
We also need to mention this on the "Breaking Changes" page [^1][^2], advising users to switch to SMTP as per https://github.com/directus/directus/issues/23217#issuecomment-2271989983.

[^1]: https://docs.directus.io/releases/breaking-changes.html
[^2]: https://github.com/directus/directus/blob/1a2a085731638bf9dff87267b3efee02d7533c2a/docs/releases/breaking-changes.md

florian-strasser committed:
Fix "frozen-lockfile" error

florian-strasser committed:
Edited breaking changes docs

florian-strasser committed:
Implented suggested change regardings changeset

rijkvanzanten committed:
Merge branch 'main' into patch-2

rijkvanzanten committed:
Merge branch 'main' into patch-2

rijkvanzanten committed:
Remove sendgrid mentions from docs

rijkvanzanten committed:
Update breaking change note

rijkvanzanten committed:
Fix spelling/formatting

rijkvanzanten committed:
One more spelling fix

u12206050 commented:
WHY?!? More importantly why not release this on a minor release, it is a big breaking change to anyone using Sendgrid.

McSundae commented:
@u12206050 I unterstand your frustration, but as mentioned in the initial issue the package that was used for this is not maintained since over 2 years, so it is a security issue to keep it inside of directus. We recommend to use smtp to connect with send grid

u12206050 commented:
> @u12206050 I understand your frustration, but as mentioned in the initial issue the package that was used for this is not maintained since over 6 years, so it is a security issue to keep it inside of directus. We recommend to use smtp to connect with send grid.
> 
> The package I'm talking about is: https://github.com/nodemailer/nodemailer-sendgrid

Thanks, that I understand, although I still don't understand how this could be released as a patch release, are we too assume from now on that patch releases could also contain breaking changes?

jogi-k commented:
Same opinion. Why are you bringing out breaking changes in hotfixes?   
Thats completely incompatible with [semantic versioning](https://semver.org/)

inovosel commented:
Have you added any guide/instructions how to get SendGrid working again after you have removed support for SendGrid transport? Or a way how to manually add SendGrid transport support (with me taking all the risks for using outdated component)?

McSundae commented:
As already mentioned it is recommended to use simply smtp. So you simply go into your config file and change the specific parameters:

https://docs.directus.io/self-hosted/config-options.html#smtp-smtp
